### ISSUE TITLE
Traces are not found when searching from the query UI

### ISSUE THREAD

**Issue by milon619** [jaegertracing/jaeger#2816]
**Describe the bug**
We are using Elasticsearch storage backend and the querier is able to discover the ES cluster and connect to it. I even tried an API request with a valid traceID which I found in the ES jaeger-span-* daily index, and its returning '[{"code":404,"msg":"trace not found"}]'

**To Reproduce**
1. Use ES as a storage backend
2. Traces are visible in Kibana
3. Search in Query UI and it's returning 404

**Expected behavior**
Search should return success and show traces

**Screenshots**
![image](https://user-images.githubusercontent.com/8054751/108185291-264fce80-7132-11eb-9d5b-6b4a47430d5d.png)
![image](https://user-images.githubusercontent.com/8054751/108185634-86467500-7132-11eb-9857-a3fd9d1d2e69.png)
![image](https://user-images.githubusercontent.com/8054751/108185804-b2fa8c80-7132-11eb-9941-2d0a8d0e385d.png)

**Version (please complete the following information):**
 - OS: Linux
 - Jaeger version: 1.21.0
 - Deployment: Kubernetes

**What troubleshooting steps did you try?**
Tried an API request with a valid traceID which I found in the ES jaeger-span-* daily index, and its returning '[{"code":404,"msg":"trace not found"}]'

**Additional context**
The situation is the same as that of https://github.com/jaegertracing/jaeger/issues/1885 , but the solution mentioned doesn't apply to us as we are not using any roll-over aliases.

cc: @jpkrohling

**Comment by jpkrohling** [comment]
@rubenvp8510, please take a look at this one as well. Seems similar to the problem you were investigating and couldn't reproduce.

**Comment by jpkrohling** [comment]
Might be related to jaegertracing/jaeger-ui#636

**Comment by milon619** [comment]
We are not even able to see today's traces.

**Comment by albertteoh** [comment]
@milon619 this is a long-shot, but can you please try dropping the leading `0` in the traceID? i.e. search on `13198...`

**Comment by milon619** [comment]
It's the same 404 error even after dropping the leading '0' in the traceID

**Comment by albertteoh** [comment]
Is this happening on _every_ traceID you've tried or only sporadically on some traceIDs?

Another check I would perform is to make the exact same ES query as jaeger-query on the ES cluster through curl/postman. I may have this query lying around somewhere and will share if I can find it.

**Comment by milon619** [comment]
This is happening for all traceIDs. It would be great if you can share the query that jaeger-query is making to ES API. Or may be some debug logging in the jaeger-query which could give some clue about it.

**Comment by albertteoh** [comment]
@milon619 I think this is an example of what jaeger-query sends to elasticsearch. Note: the endpoint I hit locally is `localhost:9200/jaeger-span-2021-02-18/_search`:

```
{
    "query": {
        "bool": {
            "should": [
                {
                    "term": {
                        "traceID": {
                            "boost": 2,
                            "value": "<your traceID>"
                        }
                    }
                },
                {
                    "term": {
                        "traceID": "<your traceID>"
                    }
                }
            ]
        }
    },
    
    "sort": [
        {
            "startTime": {
                "order": "asc"
            }
        }
    ]
}
```

Note I used `jaeger-span-2021-02-18` because I'm not enabling aliases. Also try `jaeger-span-*` if you still can't find it due to aliases or incorrect dates.

If you use the `--es.use-aliases` flag, then this could also be a cause of the "trace not found" problem, and it gets a little more complicated (see https://www.jaegertracing.io/docs/1.21/deployment/#elasticsearch-rollover):
- Reads and writes will refer to another index: `jaeger-span-read` and `jaeger-span-write`
- Both jaeger-collector and jaeger-query will need to have the same `--es.use-aliases` flags enabled
- Were the [initialization scripts](https://www.jaegertracing.io/docs/1.21/deployment/#initialize) executed
- etc.

**Comment by milon619** [comment]
Ok, let me try that using curl in our local setup and see what ES returns.

**Comment by milon619** [comment]
@albertteoh Tried the above query using curl and its returning results
`$ curl -k -XGET --header 'Content-Type: application/json' -H "Authorization:Basic xxxxxxxxxxxxxxx" https://ES_HOST/jaeger-span-2021-02-19/_search -d '
{
    "query": {
        "bool": {
            "should": [
                {
                    "term": {
                        "traceID": {
                            "boost": 2,
                            "value": "00f4f5c2d5d6bd4d"
                        }
                    }
                },
                {
                    "term": {
                        "traceID": "00f4f5c2d5d6bd4d"
                    }
                }
            ]
        }
    },

    "sort": [
        {
            "startTime": {
                "order": "asc"
            }
        }
    ]
}' | jq

{
  "took": 2147,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 12,
      "relation": "eq"
    },
    "max_score": null,
    "hits": [
      {
        "_index": "jaeger-span-2021-02-19",
        "_type": "_doc",
        "_id": "QCCeuXcB5Nc3DDmFnPm5",
        "_score": null,
        "_source": {
          "traceID": "00f4f5c2d5d6bd4d",
          "spanID": "00f4f5c2d5d6bd4d",
          "flags": 1,
          "operationName": "ingress",
          "references": [],
          "startTime": 1613726911537985,
          "startTimeMillis": 1613726911537,
          "duration": 14880,
          "tags": [
            {
              "key": "sampler.type",
              "type": "string",
              "value": "probabilistic"
            },
            {
              "key": "sampler.param",
              "type": "float64",
              "value": "0.01"
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "server"
            },
            {
              "key": "component",
              "type": "string",
              "value": "skipper"
            },
            {
              "key": "http.url",
              "type": "string",
              "value": "/cloud-api/carts-api/v1/carts/current?refresh=true"
            },
            {
              "key": "http.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "hostname",
              "type": "string",
              "value": "routing-deployment-v1-6785b599bc-x6pz5"
            },
            {
              "key": "http.remote_addr",
              "type": "string",
              "value": "10.201.171.150:15278"
            },
            {
              "key": "http.path",
              "type": "string",
              "value": "/cloud-api/carts-api/v1/carts/current"
            },
            {
              "key": "http.host",
              "type": "string",
              "value": "www.carrefour.es"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "proto"
            }
          ],
          "logs": [],
          "process": {
            "serviceName": "routing",
            "tags": [
              {
                "key": "jaeger.version",
                "type": "string",
                "value": "Go-2.25.0"
              },
              {
                "key": "hostname",
                "type": "string",
                "value": "routing-deployment-v1-6785b599bc-x6pz5"
              },
              {
                "key": "ip",
                "type": "string",
                "value": "172.22.201.119"
              },
              {
                "key": "client-uuid",
                "type": "string",
                "value": "7c03432e125749e7"
              }
            ]
          }
        },
        "sort": [
          1613726911537985
        ]
      },
      {
        "_index": "jaeger-span-2021-02-19",
        "_type": "_doc",
        "_id": "cyCeuXcB5Nc3DDmFnPnA",
        "_score": null,
        "_source": {
          "traceID": "00f4f5c2d5d6bd4d",
          "spanID": "5cb31bb4b62659ca",
          "flags": 1,
          "operationName": "request_filters",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "00f4f5c2d5d6bd4d",
              "spanID": "00f4f5c2d5d6bd4d"
            }
          ],
          "startTime": 1613726911538029,
          "startTimeMillis": 1613726911538,
          "duration": 7003,
          "tags": [
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "proto"
            }
          ],
          "logs": [],
          "process": {
            "serviceName": "routing",
            "tags": [
              {
                "key": "jaeger.version",
                "type": "string",
                "value": "Go-2.25.0"
              },
              {
                "key": "hostname",
                "type": "string",
                "value": "routing-deployment-v1-6785b599bc-x6pz5"
              },
              {
                "key": "ip",
                "type": "string",
                "value": "172.22.201.119"
              },
              {
                "key": "client-uuid",
                "type": "string",
                "value": "7c03432e125749e7"
              }
            ]
          }
        },
        "sort": [
          1613726911538029
        ]
      },
      {
        "_index": "jaeger-span-2021-02-19",
        "_type": "_doc",
        "_id": "n6ueuXcB6krxHSoMl0-w",
        "_score": null,
        "_source": {
          "traceID": "00f4f5c2d5d6bd4d",
          "spanID": "26c6341d68dbb31e",
          "flags": 1,
          "operationName": "HTTP call http://sessions-api-service-v1.core.svc.cluster.local:8080/v1/sessions",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "00f4f5c2d5d6bd4d",
              "spanID": "00f4f5c2d5d6bd4d"
            }
          ],
          "startTime": 1613726911538450,
          "startTimeMillis": 1613726911538,
          "duration": 6314,
          "tags": [
            {
              "key": "span.kind",
              "type": "string",
              "value": "client"
            },
            {
              "key": "http.url",
              "type": "string",
              "value": "HTTP call http://sessions-api-service-v1.core.svc.cluster.local:8080/v1/sessions"
            },
            {
              "key": "http.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "proto"
            }
          ],
          "logs": [],
          "process": {
            "serviceName": "routing",
            "tags": [
              {
                "key": "env",
                "type": "string",
                "value": "PRO"
              },
              {
                "key": "jaeger.version",
                "type": "string",
                "value": "Go-2.25.0"
              },
              {
                "key": "hostname",
                "type": "string",
                "value": "routing-deployment-v1-6785b599bc-x6pz5"
              },
              {
                "key": "ip",
                "type": "string",
                "value": "172.22.201.119"
              },
              {
                "key": "client-uuid",
                "type": "string",
                "value": "407a510bbbcb241d"
              }
            ]
          }
        },
        "sort": [
          1613726911538450
        ]
      },
      {
        "_index": "jaeger-span-2021-02-19",
        "_type": "_doc",
        "_id": "ZyCeuXcB5Nc3DDmFnPnA",
        "_score": null,
        "_source": {
          "traceID": "00f4f5c2d5d6bd4d",
          "spanID": "3e239b5d4738dcd6",
          "flags": 1,
          "operationName": "server",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "00f4f5c2d5d6bd4d",
              "spanID": "26c6341d68dbb31e"
            }
          ],
          "startTime": 1613726911541900,
          "startTimeMillis": 1613726911541,
          "duration": 1941,
          "tags": [
            {
              "key": "span.kind",
              "type": "string",
              "value": "server"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "proto"
            }
          ],
          "logs": [],
          "process": {
            "serviceName": "sessions-api",
            "tags": [
              {
                "key": "env",
                "type": "string",
                "value": "PRO"
              },
              {
                "key": "jaeger.version",
                "type": "string",
                "value": "Go-2.25.0"
              },
              {
                "key": "hostname",
                "type": "string",
                "value": "sessions-api-deployment-v1-575d9fc9bb-bnmws"
              },
              {
                "key": "ip",
                "type": "string",
                "value": "172.22.202.230"
              },
              {
                "key": "client-uuid",
                "type": "string",
                "value": "6c6eebd844e3e9ff"
              }
            ]
          }
        },
        "sort": [
          1613726911541900
        ]
      },
      {
        "_index": "jaeger-span-2021-02-19",
        "_type": "_doc",
        "_id": "aCCeuXcB5Nc3DDmFj_C0",
        "_score": null,
        "_source": {
          "traceID": "00f4f5c2d5d6bd4d",
          "spanID": "472c6c0b3d24770c",
          "flags": 1,
          "operationName": "GET /v1/sessions?id=1oh7KxEY3SUIYqpNdvycrDA6ZtA&login_check=true",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "00f4f5c2d5d6bd4d",
              "spanID": "3e239b5d4738dcd6"
            }
          ],
          "startTime": 1613726911541915,
          "startTimeMillis": 1613726911541,
          "duration": 1930,
          "tags": [
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "proto"
            }
          ],
          "logs": [
            {
              "timestamp": 1613726911541922,
              "fields": [
                {
                  "key": "event",
                  "type": "string",
                  "value": "baggage"
                },
                {
                  "key": "key",
                  "type": "string",
                  "value": "session_id"
                },
                {
                  "key": "value",
                  "type": "string",
                  "value": "1oh7KxEY3SUIYqpNdvycrDA6ZtA"
                }
              ]
            },
            {
              "timestamp": 1613726911542070,
              "fields": [
                {
                  "key": "event",
                  "type": "string",
                  "value": "baggage"
                },
                {
                  "key": "key",
                  "type": "string",
                  "value": "id"
                },
                {
                  "key": "value",
                  "type": "string",
                  "value": ""
                }
              ]
            },
            {
              "timestamp": 1613726911542074,
              "fields": [
                {
                  "key": "event",
                  "type": "string",
                  "value": "baggage"
                },
                {
                  "key": "key",
                  "type": "string",
                  "value": "login_check"
                },
                {
                  "key": "value",
                  "type": "string",
                  "value": ""
                }
              ]
            }
          ],
          "process": {
            "serviceName": "sessions-api",
            "tags": [
              {
                "key": "env",
                "type": "string",
                "value": "PRO"
              },
              {
                "key": "jaeger.version",
                "type": "string",
                "value": "Go-2.25.0"
              },
              {
                "key": "hostname",
                "type": "string",
                "value": "sessions-api-deployment-v1-575d9fc9bb-bnmws"
              },
              {
                "key": "ip",
                "type": "string",
                "value": "172.22.202.230"
              },
              {
                "key": "client-uuid",
                "type": "string",
                "value": "6c6eebd844e3e9ff"
              }
            ]
          }
        },
        "sort": [
          1613726911541915
        ]
      },
      {
        "_index": "jaeger-span-2021-02-19",
        "_type": "_doc",
        "_id": "kKueuXcB6krxHSoMj09N",
        "_score": null,
        "_source": {
          "traceID": "00f4f5c2d5d6bd4d",
          "spanID": "079c8f5ab31dc437",
          "flags": 1,
          "operationName": "server",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "00f4f5c2d5d6bd4d",
              "spanID": "11197354d72b0569"
            }
          ],
          "startTime": 1613726911544803,
          "startTimeMillis": 1613726911544,
          "duration": 6780,
          "tags": [
            {
              "key": "span.kind",
              "type": "string",
              "value": "server"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "proto"
            }
          ],
          "logs": [],
          "process": {
            "serviceName": "carts-api",
            "tags": [
              {
                "key": "env",
                "type": "string",
                "value": "PRO"
              },
              {
                "key": "jaeger.version",
                "type": "string",
                "value": "Go-2.25.0"
              },
              {
                "key": "hostname",
                "type": "string",
                "value": "carts-api-deployment-v1-675d4f84d5-bq4f7"
              },
              {
                "key": "ip",
                "type": "string",
                "value": "172.22.197.24"
              },
              {
                "key": "client-uuid",
                "type": "string",
                "value": "2dbc2294acfd2aa5"
              }
            ]
          }
        },
        "sort": [
          1613726911544803
        ]
      },
      {
        "_index": "jaeger-span-2021-02-19",
        "_type": "_doc",
        "_id": "7CCeuXcB5Nc3DDmFl_iS",
        "_score": null,
        "_source": {
          "traceID": "00f4f5c2d5d6bd4d",
          "spanID": "7e3c3a8903d1f71d",
          "flags": 1,
          "operationName": "GET /v1/carts/current?refresh=true",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "00f4f5c2d5d6bd4d",
              "spanID": "079c8f5ab31dc437"
            }
          ],
          "startTime": 1613726911544819,
          "startTimeMillis": 1613726911544,
          "duration": 6768,
          "tags": [
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "proto"
            }
          ],
          "logs": [
            {
              "timestamp": 1613726911544827,
              "fields": [
                {
                  "key": "event",
                  "type": "string",
                  "value": "baggage"
                },
                {
                  "key": "key",
                  "type": "string",
                  "value": "session_id"
                },
                {
                  "key": "value",
                  "type": "string",
                  "value": "1oh7KxEY3SUIYqpNdvycrDA6ZtA"
                }
              ]
            },
            {
              "timestamp": 1613726911544838,
              "fields": [
                {
                  "key": "event",
                  "type": "string",
                  "value": "baggage"
                },
                {
                  "key": "key",
                  "type": "string",
                  "value": "refresh"
                },
                {
                  "key": "value",
                  "type": "string",
                  "value": ""
                }
              ]
            }
          ],
          "process": {
            "serviceName": "carts-api",
            "tags": [
              {
                "key": "env",
                "type": "string",
                "value": "PRO"
              },
              {
                "key": "jaeger.version",
                "type": "string",
                "value": "Go-2.25.0"
              },
              {
                "key": "hostname",
                "type": "string",
                "value": "carts-api-deployment-v1-675d4f84d5-bq4f7"
              },
              {
                "key": "ip",
                "type": "string",
                "value": "172.22.197.24"
              },
              {
                "key": "client-uuid",
                "type": "string",
                "value": "2dbc2294acfd2aa5"
              }
            ]
          }
        },
        "sort": [
          1613726911544819
        ]
      },
      {
        "_index": "jaeger-span-2021-02-19",
        "_type": "_doc",
        "_id": "XCCeuXcB5Nc3DDmFj_C0",
        "_score": null,
        "_source": {
          "traceID": "00f4f5c2d5d6bd4d",
          "spanID": "7ee048df64b3a8c3",
          "flags": 1,
          "operationName": "HTTP call http://sessions-api-service-v1.core.svc.cluster.local:8080/v1/sessions",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "00f4f5c2d5d6bd4d",
              "spanID": "7e3c3a8903d1f71d"
            }
          ],
          "startTime": 1613726911544946,
          "startTimeMillis": 1613726911544,
          "duration": 4425,
          "tags": [
            {
              "key": "span.kind",
              "type": "string",
              "value": "client"
            },
            {
              "key": "http.url",
              "type": "string",
              "value": "HTTP call http://sessions-api-service-v1.core.svc.cluster.local:8080/v1/sessions"
            },
            {
              "key": "http.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "proto"
            }
          ],
          "logs": [],
          "process": {
            "serviceName": "carts-api",
            "tags": [
              {
                "key": "env",
                "type": "string",
                "value": "PRO"
              },
              {
                "key": "jaeger.version",
                "type": "string",
                "value": "Go-2.25.0"
              },
              {
                "key": "hostname",
                "type": "string",
                "value": "carts-api-deployment-v1-675d4f84d5-bq4f7"
              },
              {
                "key": "ip",
                "type": "string",
                "value": "172.22.197.24"
              },
              {
                "key": "client-uuid",
                "type": "string",
                "value": "2dbc2294acfd2aa5"
              }
            ]
          }
        },
        "sort": [
          1613726911544946
        ]
      },
      {
        "_index": "jaeger-span-2021-02-19",
        "_type": "_doc",
        "_id": "_CCeuXcB5Nc3DDmFmvh3",
        "_score": null,
        "_source": {
          "traceID": "00f4f5c2d5d6bd4d",
          "spanID": "11197354d72b0569",
          "flags": 1,
          "operationName": "proxy",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "00f4f5c2d5d6bd4d",
              "spanID": "00f4f5c2d5d6bd4d"
            }
          ],
          "startTime": 1613726911545075,
          "startTimeMillis": 1613726911545,
          "duration": 7788,
          "tags": [
            {
              "key": "span.kind",
              "type": "string",
              "value": "client"
            },
            {
              "key": "skipper.route_id",
              "type": "string",
              "value": "cartsApiCollage"
            },
            {
              "key": "skipper.route",
              "type": "string",
              "value": "Path(\"/cloud-api/carts-api/v1/carts/*\") -> logIngress() -> session() -> modPath(\"/cloud-api/carts-api/v1\", \"/v1\") -> \"http://carts-api-service-v1.business-checkout.svc.cluster.local:8080\""
            },
            {
              "key": "component",
              "type": "string",
              "value": "skipper"
            },
            {
              "key": "http.url",
              "type": "string",
              "value": "http://carts-api-service-v1.business-checkout.svc.cluster.local:8080/v1/carts/current"
            },
            {
              "key": "http.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "hostname",
              "type": "string",
              "value": "routing-deployment-v1-6785b599bc-x6pz5"
            },
            {
              "key": "http.remote_addr",
              "type": "string",
              "value": ""
            },
            {
              "key": "http.path",
              "type": "string",
              "value": "/v1/carts/current"
            },
            {
              "key": "http.host",
              "type": "string",
              "value": "carts-api-service-v1.business-checkout.svc.cluster.local:8080"
            },
            {
              "key": "http.status_code",
              "type": "int64",
              "value": "204"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "proto"
            }
          ],
          "logs": [
            {
              "timestamp": 1613726911545118,
              "fields": [
                {
                  "key": "http_roundtrip",
                  "type": "string",
                  "value": "start"
                }
              ]
            },
            {
              "timestamp": 1613726911552657,
              "fields": [
                {
                  "key": "http_roundtrip",
                  "type": "string",
                  "value": "end"
                }
              ]
            }
          ],
          "process": {
            "serviceName": "routing",
            "tags": [
              {
                "key": "jaeger.version",
                "type": "string",
                "value": "Go-2.25.0"
              },
              {
                "key": "hostname",
                "type": "string",
                "value": "routing-deployment-v1-6785b599bc-x6pz5"
              },
              {
                "key": "ip",
                "type": "string",
                "value": "172.22.201.119"
              },
              {
                "key": "client-uuid",
                "type": "string",
                "value": "7c03432e125749e7"
              }
            ]
          }
        },
        "sort": [
          1613726911545075
        ]
      },
      {
        "_index": "jaeger-span-2021-02-19",
        "_type": "_doc",
        "_id": "l6ueuXcB6krxHSoMj09N",
        "_score": null,
        "_source": {
          "traceID": "00f4f5c2d5d6bd4d",
          "spanID": "327a4d8efeb44864",
          "flags": 1,
          "operationName": "server",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "00f4f5c2d5d6bd4d",
              "spanID": "7ee048df64b3a8c3"
            }
          ],
          "startTime": 1613726911548266,
          "startTimeMillis": 1613726911548,
          "duration": 990,
          "tags": [
            {
              "key": "span.kind",
              "type": "string",
              "value": "server"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "proto"
            }
          ],
          "logs": [],
          "process": {
            "serviceName": "sessions-api",
            "tags": [
              {
                "key": "env",
                "type": "string",
                "value": "PRO"
              },
              {
                "key": "jaeger.version",
                "type": "string",
                "value": "Go-2.25.0"
              },
              {
                "key": "hostname",
                "type": "string",
                "value": "sessions-api-deployment-v1-575d9fc9bb-wbgxj"
              },
              {
                "key": "ip",
                "type": "string",
                "value": "172.22.201.140"
              },
              {
                "key": "client-uuid",
                "type": "string",
                "value": "758e4fd193682b48"
              }
            ]
          }
        },
        "sort": [
          1613726911548266
        ]
      }
    ]
  }
}`

Trace found in Kibana
![image](https://user-images.githubusercontent.com/8054751/108489759-f63a3400-72c7-11eb-8c29-8c350b4261c3.png)

**Comment by albertteoh** [comment]
Okay, we can definitely rule out ES. What do the logs show in your jaeger-query instance? I think if you're seeing a "trace not found" error, it means jaeger-query was able to successfully connect to ES with the correct auth credentials, which you did mention in the issue description as well.

Can you see a list of services in Jaeger UI? Or is it showing a 0 count?

![Screen Shot 2021-02-19 at 10 29 16 pm](https://user-images.githubusercontent.com/26584478/108499114-f0f9db00-7301-11eb-9966-c1de1d401f0d.png)

If you haven't already, have you tried running a similar setup locally or even having a basic setup working locally first like running elasticsearch (no security enabled), jaeger-collector, jaeger-agent and jaeger-query together (I can show you if you're not sure).

**Comment by milon619** [comment]
Yes, we can see the list of services in the Jaeger UI. 

Jaeger instance logs
```
W
2021/02/17 08:03:45 maxprocs: Leaving GOMAXPROCS=16: CPU quota undefined
{"level":"info","ts":1613549025.8440502,"caller":"flags/service.go:115","msg":"Mounting metrics handler on admin server","route":"/metrics"}
{"level":"info","ts":1613549025.84477,"caller":"flags/admin.go:108","msg":"Mounting health check on admin server","route":"/"}
{"level":"info","ts":1613549025.8449712,"caller":"flags/admin.go:114","msg":"Starting admin HTTP server","http-port":16687}
{"level":"info","ts":1613549025.8449967,"caller":"flags/admin.go:100","msg":"Admin server started","http-port":16687,"health-status":"unavailabl
e"}
{"level":"info","ts":1613549025.8877113,"caller":"config/config.go:172","msg":"Elasticsearch detected","version":7}
{"level":"info","ts":1613549025.8890612,"caller":"app/static_handler.go:176","msg":"watching","file":"/etc/config/ui.json"}
{"level":"info","ts":1613549025.889119,"caller":"app/static_handler.go:184","msg":"watching","dir":"/etc/config"}
{"level":"info","ts":1613549025.8894763,"caller":"healthcheck/handler.go:130","msg":"Health Check state change","status":"ready"}
{"level":"info","ts":1613549025.8895614,"caller":"app/server.go:135","msg":"Starting CMUX server","port":16686}
{"level":"info","ts":1613549025.8895056,"caller":"app/server.go:112","msg":"Starting HTTP server","port":16686}
{"level":"info","ts":1613549025.889657,"caller":"app/server.go:125","msg":"Starting GRPC server","port":16686}
```

**Comment by yurishkuro** [comment]
@milon619 please use GitHub paste for large output like the json above, inline it makes this issue difficult to follow.

@albertteoh it seems it would be very useful to log the es query, not sure if the driver allows that

**Comment by albertteoh** [comment]
@milon619 cool, please browse to a trace, get its traceID from the URL (e.g. http://localhost:16686/trace/3f51eb33d32f57d9) and enter it into the traceID Search bar to see if that works:

<img width="995" alt="Screen Shot 2021-02-21 at 6 29 08 am" src="https://user-images.githubusercontent.com/26584478/108606494-71a5fd80-740e-11eb-8eec-671bc91bbcb9.png">

@yurishkuro #2821: the client interface has some minor limitations, but it's possible.

**Comment by milon619** [comment]
@albertteoh it's not working either

![image](https://user-images.githubusercontent.com/8054751/108668560-6384d980-7501-11eb-9e75-723f643645b8.png)


![image](https://user-images.githubusercontent.com/8054751/108668480-510aa000-7501-11eb-9963-096bc5248c1a.png)

![Uploading image.png…]()

**Comment by milon619** [comment]
Hi @albertteoh , one question. How can I find the version of Jaeger Query instance ? The UI looks very different in our setup than the one in your screenshot.

**Comment by albertteoh** [comment]
@milon619 sorry I wasn't clear. I did not mean browse to a trace in Kibana, but browse to a trace using Jaeger UI:
- Select a service like "Routing"
- Click "Find Traces"
![Screen Shot 2021-02-22 at 9 12 37 pm](https://user-images.githubusercontent.com/26584478/108694223-e41ff600-7552-11eb-95b6-e76c71012e22.png)
- Click on a trace:
![Screen Shot 2021-02-22 at 9 14 33 pm](https://user-images.githubusercontent.com/26584478/108694323-00239780-7553-11eb-8471-9c932d30a1b9.png)
- Copy the traceID from the URL as in my earlier comment and paste it into the search bar.

> How can I find the version of Jaeger Query instance ?

Top right corner:

![Screen Shot 2021-02-22 at 9 10 20 pm](https://user-images.githubusercontent.com/26584478/108694190-dbc7bb00-7552-11eb-97a8-b7d2be5c1fd0.png)

**Comment by milon619** [comment]
@albertteoh what you have asked for is what is not working for us and that's why we opened this issue. Sorry if I misunderstood you.

**Comment by milon619** [comment]
![image](https://user-images.githubusercontent.com/8054751/108697110-6d6e0300-7528-11eb-93bb-c8c228ee1a11.png)

**Comment by albertteoh** [comment]
It's odd that you're seeing Services and Operations but no trace data. It's possible there's not data for `routing`.

Have you tried changing "Service" to `header-papi` (I noticed in your kibana screenshot) and increase the "Lookback" to a couple of days?

FWIW, here's the query that is sent from jaeger-query to Elasticsearch when you click on "Find Traces". I tested it out and looks to be returning a collection of TraceIDs. You'd of course, need to modify the time range, and serviceName:
```
GET localhost:9200/jaeger-span-*/_search 
{
  "size": 0,
  "query": {
    "bool": {
      "must": [
        {
          "range": {
            "startTime": {
              "from": 1614077800337000,
              "to": 1614081400337000,
              "include_lower": true,
              "include_upper": true,
              "boost": 1
            }
          }
        },
        {
          "match": {
            "process.serviceName": {
              "query": "jaeger-query",
              "operator": "OR",
              "prefix_length": 0,
              "max_expansions": 50,
              "fuzzy_transpositions": true,
              "lenient": false,
              "zero_terms_query": "NONE",
              "auto_generate_synonyms_phrase_query": true,
              "boost": 1
            }
          }
        }
      ],
      "adjust_pure_negative": true,
      "boost": 1
    }
  },
  "track_total_hits": 2147483647,
  "aggregations": {
    "traceIDs": {
      "terms": {
        "field": "traceID",
        "size": 20,
        "min_doc_count": 1,
        "shard_min_doc_count": 0,
        "show_term_doc_count_error": false,
        "order": [
          {
            "startTime": "desc"
          },
          {
            "_key": "asc"
          }
        ]
      },
      "aggregations": {
        "startTime": {
          "max": {
            "field": "startTime"
          }
        }
      }
    }
  }
}
```

**Comment by jpkrohling** [comment]
@milon619 how critical is your tracing data? Would you be OK with sharing a dump of your ES privately with me, so that I can try to debug your problem further?

**Comment by milon619** [comment]
@albertteoh / @jpkrohling it turned out that we were running an old version of Jaeger querier image in our deployment and after updating it to 1.21 we are able to see again the traces in the UI. However, the strange thing is with the old version everything was working fine and suddenly one day it stopped working. @albertteoh maybe that explains why in the UI we're seeing Services and Operations but no trace data.

**Comment by milon619** [comment]
Closing this issue

**Comment by jpkrohling** [comment]
> it turned out that we were running an old version of Jaeger querier image in our deployment and after updating it to 1.21 we are able to see again the traces in the UI

I can't think of a reason for this to happen. @albertteoh any ideas?

**Comment by albertteoh** [comment]
Neither can I, I haven't come across a situation where services/operations are queryable but spans not; though technically possible because they reside in separate indices (`jaeger-service-*` and `jaeger-span-*`). 

In this case, the `jaeger-span-*` index may have been corrupted, and updating jaeger may have, _somehow_ (I can't think of how), fixed the index. However, I would have thought version 1.19 would have been able to do this (fix the index) when rolling over to a new day anyway; unless if the index template was corrupt as well... but at this point, I'm just pulling at straws :)

**Comment by ndrscodes** [comment]
Could we reopen this? We are currently having this issue using jaeger 1.24.0. Traces can be searched using the jaeger search, but, using the Trace ID lookup, they only show the 404 error page.
Curiously though, the URL for the error page and the successful trace lookup page is the same: https://cool-tracing-url.local/trace/1492ecb6434b553e, With one showing all spans and the other showing an API 404 error page.

**Comment by jpkrohling** [comment]
@ndrscodes, could you please try the `latest` image (recent as of 8 minutes ago), or perhaps 1.25? I think @pavolloffay fixed an issue that might influence this one.

**Comment by ndrscodes** [comment]
@jpkrohling i'll update our system today. I'll check if it's fixed ASAP! Thanks :)

**Comment by ndrscodes** [comment]
@jpkrohling i should mention that we are currently (as stupid as it may sound) running jaeger all-in-one as a simple process on a server, so i can't check the latest image. I did update our staging system to jaeger 1.25.0 though. Sadly, the problem still persists. 
I've found a way to fix the search (at least in one browser tab): If i use the "manual" search (not searching by trace id) by selecting the service in question and specifying a timeframe close to the trace id i'm searching for, after hitting "find traces" and waiting for the search to complete, i can use the trace id lookup for traces within the timeframe i just searched. When i copy the trace id's URL which i just found successfully through the trace lookup and paste it in another browser window, it returns 404 again.
